fdbserver/CommitProxyServer.actor.cpp

```diff
@@ -1056,6 +1064,31 @@ ACTOR Future<Void> assignMutationsToStorageServers(CommitBatchContext* self) {
 	return Void();
 }
 
+// Send the version and the set of tags we are writing to to the version indexers. This can be made asynchronous in the
+// future, but for now it isn't in order to simplify the implementation.
+Future<Void> informVersionIndexers(CommitBatchContext* self) {
+	VersionIndexerCommitRequest req;
+	req.version = self->commitVersion;
+	req.previousVersion = self->prevVersion;
+	req.committedVersion = self->pProxyCommitData->committedVersion.get();
+	req.tags.reserve(self->writtenTags.size());
+	req.tags.insert(req.tags.end(), self->writtenTags.begin(), self->writtenTags.end());
+	std::vector<Future<Void>> resp;
+	const auto& versionIndexers = self->pProxyCommitData->db->get().versionIndexers;
+	resp.reserve(versionIndexers.size());
+	for (const auto& vi : versionIndexers) {
+		resp.emplace_back(vi.commit.getReply(req));
+	}
+	return waitForAll(resp);
+}
+
+// A helper function. We need to wait on logging and on the version indexers to acknowledge the write. However, we are
+// only interested in the result from the tlogs.
+ACTOR Future<Version> loggingComplete(Future<Version> logging, Future<Void> versionIndexersInformed) {
+	wait(versionIndexersInformed && success(logging));
```

Comment:
in this line, will the messages to the version indexer and logs be sent in parallel ?

Reply:
Yes, they will happen in parallel. I don't think we would strictly need to wait on the version indexer here, but for now this is simpler (and the version indexer is expected to reply much earlier as it doesn't have to write to disk).